packages/core/src/pipelines/event.pipeline.ts

```diff
@@ -0,0 +1,152 @@
+        }
+
+        return interceptedEvent;
+    }
+
+    private async postMappingIntercept(event: Event<any>): Promise<Event<any>> {
+        let interceptedEvent = event;
+
+        for (const eventInterceptor of this.eventInterceptors) {
+            interceptedEvent = await eventInterceptor.postMappingIntercept?.(interceptedEvent) ?? interceptedEvent;
+        }
+
+        return interceptedEvent;
+    }
+
+    private async preResponseMappingIntercept(eventResponse: EventResponse<any, any>): Promise<EventResponse<any, any>> {
+        let interceptedEventResponse = eventResponse;
+
+        for (const eventInterceptor of this.eventInterceptors) {
+            interceptedEventResponse = await eventInterceptor.preResponseMappingIntercept?.(interceptedEventResponse) ?? interceptedEventResponse;
+        }
+
+        return interceptedEventResponse;
+    }
+
+    private async postResponseMappingIntercept(eventResponse: object): Promise<object> {
+        let interceptedEventResponse = eventResponse;
+
+        for (const eventInterceptor of this.eventInterceptors) {
+            interceptedEventResponse = await eventInterceptor.postResponseMappingIntercept?.(interceptedEventResponse) ?? interceptedEventResponse;
+        }
+
+        return interceptedEventResponse;
+    }
+
+    private async executeEvent(event: Event<any>, eventDispatcher: EventDispatcherInterface): Promise<EventResponse<any, any>> {
+        // 1 - Run the post mapped interceptors on every single event before they get executed.
+        let interceptedEvent = await this.postMappingIntercept(event)
+
+        // 2 - Call the EventDispatcher and retrieve the Event Response
+        return eventDispatcher.dispatch(interceptedEvent);
+    }
+
+    async execute(event: object, executionContext: ExecutionContextInterface<any>, container: DependencyContainer): Promise<any> {
+        // 1- We have the raw event, we start by executing the PreMapping Interceptors
+        const interceptedEvent = await this.preMappingIntercept(event, executionContext);
+
+        // 2- With the intercepted raw event, run the Events Mapping to get all the Events and the EventsExecutionOptions.
+        // For each event mapper that supports the event, we batch the executions for each mapper. So it's possible to execute the same
+        // event twice. This is up to the EventMappers to properly identify when they map or don't map an event. Pristine
+        // isn't responsible to determine if two events are executed twice, so be careful.
+
+        const eventExecutions: EventsExecutionOptionsInterface<any>[] = [];
+
+        let numberOfEventMappers = 0;
+
+        this.eventMappers.forEach( eventMapper => {
+            if(eventMapper.supportsMapping(interceptedEvent, executionContext)) {
+                eventExecutions.push(eventMapper.map(interceptedEvent, executionContext));
```

Comment:
If multiple mappers support the event we will have execute the event multiple times ? I'm not sure this is intuitive behaviour.

Reply:
I'm not sure what the ideal solution to that is. Technically it shouldn't, but that's what I said in the comments above. I'm not sure that limiting to one EventExecutionOptions per Event is necessarily the best idea since we would throw if more than one mapper support the same rawEvent?

Reply:
Let's keep it like that for now then